src/components/Modals/Send/hooks/useSendDetails/useSendDetails.tsx

```diff
@@ -112,35 +113,52 @@ export const useSendDetails = (): UseSendDetailsReturnType => {
   const { assetReference } = fromAssetId(assetId)
   const contractAddress = tokenOrUndefined(assetReference)
 
-  const estimateFormFees = useCallback((): Promise<FeeDataEstimate<ChainId>> => {
-    if (!asset) throw new Error('No asset found')
-
-    const { assetId, amountCryptoPrecision, to, sendMax } = getValues()
-    if (!wallet) throw new Error('No wallet connected')
-    return estimateFees({
+  const estimateFormFees = useCallback(
+    ({
       amountCryptoPrecision,
-      assetId,
-      to,
-      sendMax,
-      accountId,
-      contractAddress,
-    })
-  }, [accountId, asset, contractAddress, getValues, wallet])
+    }: {
+      amountCryptoPrecision: string
+    }): Promise<FeeDataEstimate<ChainId>> => {
+      console.log({ sendMax })
+      if (!asset) throw new Error('No asset found')
+
+      if (!wallet) throw new Error('No wallet connected')
+      return estimateFees({
+        amountCryptoPrecision,
+        assetId,
+        to,
+        sendMax,
+        accountId,
+        contractAddress,
+      })
+    },
+    [accountId, asset, assetId, contractAddress, sendMax, to, wallet],
+  )
 
-  const debouncedSetEstimatedFormFees = useMemo(() => {
-    return debounce(
-      async () => {
-        const { amountCryptoPrecision } = getValues()
-        if (amountCryptoPrecision === '') return
-        if (!asset || !accountId) return
-        const estimatedFees = await estimateFormFees()
+  // * Determines the form's state from debounced input
+  // * Valid inputs:
+  // * - Non-empty numeric values including zero
+  // * Error states:
+  // * - Insufficient funds - give > have
+  // * - Empty amount - input = ''
+  // * - Not enough native token - gas > have
+  const setEstimatedFormFeesQueryFn = useCallback(
+    async ({ queryKey }: { queryKey: [string, { amountCryptoPrecision: string }] }) => {
+      const [, { amountCryptoPrecision }] = queryKey
 
-        const hasValidBalance = cryptoHumanBalance.gte(amountCryptoPrecision)
+      if (amountCryptoPrecision === '') return null
+      if (!asset || !accountId) return null
 
+      const hasValidBalance = cryptoHumanBalance.gte(amountCryptoPrecision)
+
+      try {
         if (!hasValidBalance) {
           throw new Error('common.insufficientFunds')
         }
```

Comment:
In this PR a try is added so when we throw here its caught and then the same error is thrown in the catch. We can move this up a line outside the `try` to avoid this.

Reply:
🧠 46d533f607